embedx/embedx_test.go

```diff
@@ -0,0 +1,69 @@
+package embedx
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/ory/jsonschema/v3"
+)
+
+func TestAddSchemaResources(t *testing.T) {
+
+	t.Run("case=validate schema resources are loading schema id correctly", func(t *testing.T) {
+		c := jsonschema.NewCompiler()
+		assert.NoError(t, AddSchemaResources(c, Config, IdentityMeta, IdentityExtension))
+
+		assert.EqualValues(t, "https://github.com/ory/kratos/embedx/config.schema.json", Config.GetSchemaID())
+		assert.EqualValues(t, "ory://identity-meta", IdentityMeta.GetSchemaID())
+		assert.EqualValues(t, "ory://identity-extension", IdentityExtension.GetSchemaID())
+	})
+
+	t.Run("case=add config schema resources", func(t *testing.T) {
+		c := jsonschema.NewCompiler()
+		assert.NoError(t, AddSchemaResources(c, Config))
+
+		_, err := c.Compile(Config.GetSchemaID())
+		assert.NoError(t, err)
+	})
+
+	t.Run("case=add identity schema resources", func(t *testing.T) {
+		c := jsonschema.NewCompiler()
+		assert.NoError(t, AddSchemaResources(c, IdentityMeta))
+
+		_, err := c.Compile(IdentityMeta.GetSchemaID())
+		assert.NoError(t, err)
+	})
+
+	t.Run("case=add multiple schema resources", func(t *testing.T) {
+		c := jsonschema.NewCompiler()
+		assert.NoError(t, AddSchemaResources(c, Config, IdentityMeta))
+
+		_, err := c.Compile(IdentityMeta.GetSchemaID())
+		assert.NoError(t, err)
+
+		_, err = c.Compile(Config.GetSchemaID())
+	})
+
+	t.Run("case=dynamic multiple schema resourcces", func(t *testing.T) {
+		c := jsonschema.NewCompiler()
+
+		assert.NoError(t, AddSchemaResources(c, Config, IdentityExtension, IdentityMeta))
+
+		_, err := c.Compile(Config.GetSchemaID())
+		assert.NoError(t, err)
+
+		_, err = c.Compile(IdentityExtension.GetSchemaID())
+		assert.NoError(t, err)
+
+		_, err = c.Compile(IdentityMeta.GetSchemaID())
+		assert.NoError(t, err)
+	})
```

Comment:
Make this a table test?
```suggestion
    for _, tc := range []struct {
		desc            string
		add             []SchemaType
		expectExtraDeps []SchemaType
	}{
		{
			desc: "config schema",
			add:  []SchemaType{Config},
		},
		{
			desc:            "identity meta incl dependency",
			add:             []SchemaType{IdentityMeta},
			expectExtraDeps: []SchemaType{IdentityExtension},
		},
		{
			desc:            "multiple schemas",
			add:             []SchemaType{Config, IdentityMeta},
			expectExtraDeps: []SchemaType{IdentityExtension},
		},
		{
			desc: "double-add",
			add:  []SchemaType{IdentityExtension, IdentityMeta},
		},
	} {
		t.Run("case="+tc.desc, func(t *testing.T) {
			c := jsonschema.NewCompiler()
			assert.NoError(t, AddSchemaResources(c, tc.add...))

			for _, s := range append(tc.add, tc.expectExtraDeps...) {
				_, err := c.Compile(s.GetSchemaID())
				assert.NoError(t, err)
			}
		})
	}
```